aodndata/aatams/aatams_sattag.py

```diff
@@ -0,0 +1,171 @@
+"""A generic AATAMS_SATTAG handler."""
+import os
+import logging
+from functools import partial
+from aodncore.pipeline import (
+    HandlerBase,
+    PipelineFilePublishType,
+    FileType,
+    PipelineFile,
+)
+from aodncore.pipeline.files import RemotePipelineFileCollection
+from aodncore.pipeline.exceptions import InvalidFileContentError
+
+from .aatams_sattag_qc_dm_schema import (
+    AatamsSattagQcDmSchema,
+)  # schema is the same as dm
+
+
+NRT_FILE_REMOVAL_MSG = "NRT file %s schedule to %s"
+NRT_TIMESTAMP_COMPARISON_MSG = (
+    "NRT update requested: Comparing timestamps in the metadata files %s and %s"
+)
+NRT_TIMESTAMP_DIFFERS_MSG = (
+    "File %s within %s contains older entries than current NRT state from %s"
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def aatams_sattag_qc_dm_dest_path(filepath):
+    """Return the full filepath string for Delay files."""
+    return os.path.join(AATAMS_SATTAG_QC_DM_BASE, os.path.basename(filepath))
+
+
+def aatams_sattag_qc_nrt_dest_path(filepath):
+    """Return the full filepath string for NRT files."""
+    return os.path.join(AATAMS_SATTAG_QC_NRT_BASE, os.path.basename(filepath))
+
+
+class AatamsSattagHandler(HandlerBase):
+    """The class for aatams sattag zip/csv files."""
+
+    @staticmethod
+    def get_metadata_file(file_list):
+        """Return metadata.csv PipelineFile object within a PipelineFileCollection."""
+        fname = [x for x in file_list if "metadata" in x.name]
+        if fname:
+            return fname[0]
+        else:
+            return None
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the class with schema validation."""
+        super(AatamsSattagHandler, self).__init__(*args, **kwargs)
+        self.schema = AatamsSattagQcDmSchema()
+        self.validation_call = self.schema.extensive_validation
+        # Use below to just validate the file names and csv headers
+        # self.validation_call = self.schema.quick_validation
+
+    def is_nrt_update_required(self, old_metadata_file):
+        """Check if updates to the dataset/archive are required by inspecting the most
+        recent dates between the new state and remote(current) metadata state."""
+        new_valid_data = self.schema.metadata
+        new_recent_date = sorted(new_valid_data["qc_end_date"])[-1]
+
+        _, old_valid_data = self.schema.validate_file(old_metadata_file)
+        old_recent_date = sorted(old_valid_data["qc_end_date"])[-1]
+        return new_recent_date >= old_recent_date
+
+    def nrt_aware(self):
+        """Check if class is initialize for NRT type of files."""
+        for regex_string in self.allowed_dest_path_regexes:
+            if AATAMS_SATTAG_QC_NRT_BASE not in regex_string:
+                return False
+        return True
+
+    def schedule_file_removal(self, remote_pipeline_file):
```

Comment:
This is just a function to remove a file via pipelinepublishtype (schedule because you don't remove in place, but later). This was present before and may be required in the future, so I ought to keep it